File: plugins/aref/fixtures/go/godfile.go
// Package fixture is an intentionally over-sized package mixing concerns for aref self-testing.
package fixture

import "fmt"

type User struct {
	ID    string
	Email string
	Roles []string
}

type Item struct {
	SKU   string
	Qty   int
	Price float64
}

type Order struct {
	ID     string
	UserID string
	Items  []Item
	Total  float64
}

func Authenticate(token string) *User {
	if token == "" || len(token) < 10 || token[:7] != "bearer-" {
		return nil
	}
	raw := token[7:]
	if raw == "" || len(raw) >= 500 {
		return nil
	}
	parts := splitDot(raw)
	if len(parts) != 3 {
		return nil
	}
	for _, p := range parts {
		if p == "" {
			return nil
		}
	}
	return &User{ID: parts[1], Email: "user@example.com", Roles: []string{"user"}}
}

func splitDot(s string) []string {
	var out []string
	start := 0
	for i := 0; i < len(s); i++ {
		if s[i] == '.' {
			out = append(out, s[start:i])
			start = i + 1
		}
	}
	out = append(out, s[start:])
	return out
}

func Authorize(user *User, action string) bool {
	if user == nil {
		return false
	}
	switch action {
	case "read":
		return true
	case "write":
		return hasRole(user, "user") || hasRole(user, "admin")
	case "delete":
		return hasRole(user, "admin") || hasRole(user, "owner")
	case "admin":
		return hasRole(user, "admin")
	}
	return false
}

func hasRole(u *User, role string) bool {
	for _, r := range u.Roles {
		if r == role {
			return true
		}
	}
	return false
}

func CalculateTotal(order Order) float64 {
	total := 0.0
	for _, item := range order.Items {
		if item.Qty > 0 && item.Price > 0 {
			total += float64(item.Qty) * item.Price
		}
	}
	if len(order.Items) > 5 && total > 100 {
		total *= 0.9
	}
	return total
}

func FormatCurrency(amount float64, currency string) string {
	switch currency {
	case "USD":
		return fmt.Sprintf("$%.2f", amount)
	case "EUR":
		return fmt.Sprintf("€%.2f", amount)
	case "GBP":
		return fmt.Sprintf("£%.2f", amount)
	case "JPY":
		return fmt.Sprintf("¥%d", int(amount))
	}
	return fmt.Sprintf("%.2f %s", amount, currency)
}

File: plugins/aref/fixtures/rust/src/god_file.rs
//! Intentionally over-sized module mixing concerns for aref self-testing.

#[derive(Debug, Clone)]
pub struct User {
    pub id: String,
    pub email: String,
    pub roles: Vec<String>,
}

#[derive(Debug, Clone)]
pub struct Item {
    pub sku: String,
    pub qty: u32,
    pub price: f64,
}

#[derive(Debug, Clone)]
pub struct Order {
    pub id: String,
    pub user_id: String,
    pub items: Vec<Item>,
    pub total: f64,
}

pub fn authenticate(token: &str) -> Option<User> {
    if token.is_empty() || token.len() < 10 || !token.starts_with("bearer-") {
        return None;
    }
    let raw = &token[7..];
    if raw.is_empty() || raw.len() >= 500 || !raw.contains('.') {
        return None;
    }
    let parts: Vec<&str> = raw.split('.').collect();
    if parts.len() != 3 || parts.iter().any(|p| p.is_empty()) {
        return None;
    }
    Some(User {
        id: parts[1].to_string(),
        email: "user@example.com".to_string(),
        roles: vec!["user".to_string()],
    })
}

pub fn authorize(user: Option<&User>, action: &str) -> bool {
    let Some(u) = user else { return false };
    match action {
        "read" => true,
        "write" => u.roles.iter().any(|r| r == "user" || r == "admin"),
        "delete" => u.roles.iter().any(|r| r == "admin" || r == "owner"),
        "admin" => u.roles.iter().any(|r| r == "admin"),
        _ => false,
    }
}

pub fn calculate_total(order: &Order) -> f64 {
    let mut total = 0.0;
    for item in &order.items {
        if item.qty > 0 && item.price > 0.0 {
            total += item.qty as f64 * item.price;
        }
    }
    if order.items.len() > 5 && total > 100.0 {
        total *= 0.9;
    }
    total
}

pub fn format_currency(amount: f64, currency: &str) -> String {
    match currency {
        "USD" => format!("${:.2}", amount),
        "EUR" => format!("€{:.2}", amount),
        "GBP" => format!("£{:.2}", amount),
        "JPY" => format!("¥{}", amount.round() as i64),
        _ => format!("{:.2} {}", amount, currency),
    }
}

pub fn format_user_handle(user: &User) -> String {
    let handle = user.email.split('@').next().unwrap_or("");
    if user.roles.iter().any(|r| r == "admin") {
        format!("@{} (admin)", handle)
    } else if user.roles.iter().any(|r| r == "owner") {
        format!("@{} (owner)", handle)
    } else {
        format!("@{}", handle)
    }
}
